tasks/utils.py

```diff
@@ -376,19 +376,54 @@ def query_version(ctx, git_sha_length=7, prefix=None, major_version_hint=None):
     return version, pre, commit_number, git_sha, pipeline_id
 
 
+def cache_version(ctx, git_sha_length=7, prefix=None):
+    """
+    Generate a json cache file containing all needed variables used by get_version.
+    """
+    packed_data = {}
+    for maj_version in ['6', '7']:
+        version, pre, commits_since_version, git_sha, pipeline_id = query_version(
+            ctx, git_sha_length, prefix, major_version_hint=maj_version
+        )
+        packed_data[maj_version] = [version, pre, commits_since_version, git_sha, pipeline_id]
+    packed_data["nightly"] = is_allowed_repo_nightly_branch(os.getenv("BUCKET_BRANCH"))
+    with open(AGENT_VERSION_CACHE_NAME, "w") as file:
+        json.dump(packed_data, file, indent=4)
+
+
 def get_version(
     ctx, include_git=False, url_safe=False, git_sha_length=7, prefix=None, major_version='7', include_pipeline_id=False
 ):
-    # we only need the git info for the non omnibus builds, omnibus includes all this information by default
-
     version = ""
-    version, pre, commits_since_version, git_sha, pipeline_id = query_version(
-        ctx, git_sha_length, prefix, major_version_hint=major_version
-    )
+    pipeline_id = os.getenv("CI_PIPELINE_ID")
+    if pipeline_id and pipeline_id.isdigit():
+        try:
+            if not os.path.exists(AGENT_VERSION_CACHE_NAME):
+                ctx.run(
+                    f"aws s3 cp s3://dd-ci-artefacts-build-stable/datadog-agent/{pipeline_id}/{AGENT_VERSION_CACHE_NAME} . >/dev/null"
```

Comment:
That's a really good point didn't think about windows, at some point the output of command caused some issues that's why I removed it but now that's inside a fork it should be fine. o7

Reply:
@hush-hush Ok I remember why I have to keep it, when a CI test does ``$(inv agent.version --url-safe --major-version 7)`` for exemple. It will redirect the output of the aws s3 download as well inside of it. I'll try to redirect this command's output without ``/dev/null`` to be windows-compatible but sadly I can't show the output log for that reason.

Reply:
You have the `--quiet` option on `s3 cp`.